src/components/sidebar/__internal__/sidebar-header/sidebar-header.component.tsx

```diff
@@ -1,18 +1,34 @@
 import React from "react";
+import { PaddingProps } from "styled-system";
+import StyledSidebarHeader from "./sidebar-header.style";
 
-import SidebarHeaderStyle from "./sidebar-header.style";
-
-interface SidebarHeaderProps {
+export interface SidebarHeaderProps extends PaddingProps {
   /** This component supports children. */
   children?: React.ReactNode;
   /** A custom id. */
   id: string;
+  /** Close icon button to be rendered */
+  closeIcon?: React.ReactNode;
 }
 
-const SidebarHeader = ({ children, id }: SidebarHeaderProps) => (
-  <SidebarHeaderStyle id={id} data-component="sidebar-header">
+const SidebarHeader = ({
+  children,
+  id,
+  closeIcon,
+  ...rest
+}: SidebarHeaderProps) => (
+  <StyledSidebarHeader
+    hasClose={!!closeIcon}
+    id={id}
+    data-component="sidebar-header"
+    p="27px 32px 32px"
```

Comment:
Would we be able to specify the padding in the `style.ts` file instead?

Reply:
It's possible to move it back to the style file and have it work. I moved it to avoid having the padding style applied twice with one overwritten, basically to avoid convoluting the styles unnecessarily (albeit this time the styles aren't massively so already)

with default in style file: 
![image](https://user-images.githubusercontent.com/44157880/204556115-94e07845-18d9-4588-8792-1a88afad2fc5.png)

with default set in component:
![image](https://user-images.githubusercontent.com/44157880/204556298-d44c178b-5621-4088-ae47-709874c24c2f.png)